polytiles/polytiles.py

```diff
@@ -476,38 +468,63 @@ def generate(self, queue):
                         queue.put(t)
 
 
-def render_tiles_multithreaded(generator, mapfile, writer, num_threads=2,
-                               scale=1.0, renderlist=False):
-    logging.info("render_tiles_multithreaded(%s %s %s %s)", generator, mapfile, writer, num_threads)
+class MapnikRenderer:
+    def __init__(self, mapfile, scale):
+        self.mapfile = mapfile
+        self.scale = scale
+        self.scaled_size = int(TILE_SIZE * scale)
+        self.m = mapnik.Map(self.scaled_size, self.scaled_size)
+        mapnik.load_map(self.m, self.mapfile, True)
+        self.prj = mapnik.Projection(self.m.srs)
+
+    def get_map(self):
+        return self.m
+
+    def get_mapfile(self):
+        return self.mapfile
+
+    def get_projection(self):
+        return self.prj
+
+    def get_scale(self):
+        return self.scale
+
+    def get_scaled_size(self):
+        return self.scaled_size
+
+
+def render_tiles_multithreaded(generator, maprenderer, writer, num_threads=2, scale=1.0, renderlist=False):
+    logging.info("render_tiles_multithreaded(%s %s %s %s)", generator, maprenderer.get_mapfile(), writer, num_threads)
     printLock = multiprocessing.Lock()
     queue = multiprocessing.JoinableQueue(32)
+
     renderers = {}
     for i in range(num_threads):
-        renderer = RenderThread(writer, mapfile, queue, printLock,
-                                scale=scale, renderlist=renderlist)
+        renderer = RenderThread(writer, maprenderer, queue, printLock, scale=scale, renderlist=renderlist)
```

Comment:
The reason for passing the `Mapnik` object initialization to individual threads was that the object was not thread-safe, and there could be side effects. Did you test that the new way works? E.g. try with 32 threads and big enough map.